### PR TITLE
User's page loading and empty states

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# [User's page loading and empty states](https://app.shortcut.com/oazo-apps/story/7579/rebuild-owner-s-page)

https://user-images.githubusercontent.com/16230404/213686647-0f18cada-3be4-4a5e-8916-ac239a31e333.mp4

## Changes 👷‍♀️

- Added skeleton loading states to assets chart, positions list and vault suggestions,
- added empty states for asset chart and positions list,
- redid positions list to use discover table,
- refined some interfaces and html structure to be shorter and easier to read,
- fixed animation of product cards appearing on list (it was broken since I joined oasis 😬),
- added some more options to skeleton component - grid and doughnut.
  
## How to test 🧪

- Turn on `FollowVaults` feature flag - without it there are no lading states at all,
- testing scenarios that has to be covered:
  - wallet that has positions and one that doesn't,
  - wallet that has any assets in it or doesn't,
  - all above from perspective of an owner, connected guest and not being connected at all - depending on user's state there are different things visible and different messages.

### CODE REVIEW

File: features/discover/common/DiscoverTableDataCellContent.tsx [hunk 1]
```diff
@@ -120,6 +141,28 @@ export function DiscoverTableDataCellContent({
           )}
         </>
       )
+    case 'protection':
+      return (
+        <>
+          {primitives.cdpId && primitives[label] >= 0 ? (
```
Comment (piekczyk) [inline features/discover/common/DiscoverTableDataCellContent.tsx:147]: by design we know that `privitives[label]` will be always of type number here?

File: features/vaultsOverview/containers/PositionsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  const checksumAddress = getAddress(address.toLocaleLowerCase())
+  const { ownersPositionsList$ } = useAppContext()
+  const { walletAddress } = useAccount()
+  const [ownersPositionsListData, ownersPositionsListError] = useObservable(
+    ownersPositionsList$(checksumAddress),
+  )
+
+  const isOwner = address === walletAddress
+
+  console.log(ownersPositionsListData)
```
Comment (piekczyk) [inline features/vaultsOverview/containers/PositionsTable.tsx:38]: remove

File: features/vaultsOverview/containers/AssetsAndPositionsOverview.tsx [hunk 3]
```diff
@@ -281,54 +293,60 @@ function AssetsAndPositionsView(props: TopAssetsAndPositionsViewModal) {
...
           <Box sx={{ flexGrow: 1 }}>
             <Text
               variant="paragraph2"
               sx={{
                 fontWeight: 'semiBold',
               }}
             >
-              {t('vaults-overview.assets-and-positions', { number: topAssetsAndPositions.length })}
+              {t('vaults-overview.assets-and-positions', {
+                number: topAssetsAndPositions.length ? topAssetsAndPositions.length : '',
```
Comment (piekczyk) [inline features/vaultsOverview/containers/AssetsAndPositionsOverview.tsx:310]: `topAssetsAndPositions.length || ''`

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: features/discover/common/DiscoverTableDataCellContent.tsx (hunk 1 above)
Comment (piotrkonowrocki) [reply features/discover/common/DiscoverTableDataCellContent.tsx:147]: Yes, in that case `privitives[label]` is equivalent of `primitives.protection` and it always comes as number.

File: features/vaultsOverview/containers/PositionsTable.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply features/vaultsOverview/containers/PositionsTable.tsx:38]: 😬

File: features/vaultsOverview/containers/AssetsAndPositionsOverview.tsx (hunk 3 above)
Comment (piotrkonowrocki) [reply features/vaultsOverview/containers/AssetsAndPositionsOverview.tsx]: Fixed.

File: features/vaultsOverview/containers/PositionsTable.tsx (hunk 2 above)
Comment (piotrkonowrocki) [reply features/vaultsOverview/containers/PositionsTable.tsx]: Fixed.

**Review by piekczyk — APPROVED** [review]

**Pull request merged**